dot/parachain/candidate-validation/candidate_validation_test.go

```diff
@@ -677,3 +669,215 @@ func Test_performBasicChecks(t *testing.T) {
+		expectedError error
+		isValid       bool
+	}{
+		"invalid_pov_hash": {
+			args: args{
+				runtimeInstance:  mockInstance,
+				candidateReceipt: candidateReceipt2,
+				pov:              pov,
+			},
+			want: &ValidationResult{
+				InvalidResult: &povHashMismatch,
+			},
+			isValid: false,
+		},
+		"invalid_pov_size": {
+			args: args{
+				runtimeInstance:  mockInstance,
+				candidateReceipt: candidateReceipt3,
+				pov:              pov,
+			},
+			want: &ValidationResult{
+				InvalidResult: &paramsTooLarge,
+			},
+		},
+		"code_mismatch": {
+			args: args{
+				runtimeInstance:  mockInstance,
+				candidateReceipt: candidateReceipt4,
+				pov:              pov,
+			},
+			want: &ValidationResult{
+				InvalidResult: &codeHashMismatch,
+			},
+			isValid: false,
+		},
+		"bad_signature": {
+			args: args{
+				runtimeInstance:  mockInstance,
+				candidateReceipt: candidateReceipt5,
+				pov:              pov,
+			},
+			want: &ValidationResult{
+				InvalidResult: &badSignature,
+			},
+			isValid: false,
+		},
+		"happy_path": {
+			args: args{
+				runtimeInstance:  mockInstance,
+				candidateReceipt: candidateReceipt,
+				pov:              pov,
+			},
+			want: &ValidationResult{
+				ValidResult: &ValidValidationResult{
+					CandidateCommitments: parachaintypes.CandidateCommitments{
+						UpwardMessages:     nil,
+						HorizontalMessages: nil,
+						NewValidationCode:  nil,
+						HeadData: parachaintypes.HeadData{Data: []byte{2, 0, 0, 0, 0, 0, 0, 0, 123, 207, 206, 8, 219, 227,
+							136, 82, 236, 169, 14, 100, 45, 100, 31, 177, 154, 160, 220, 245, 59, 106, 76, 168, 122, 109,
+							164, 169, 22, 46, 144, 39, 103, 92, 31, 78, 66, 72, 252, 64, 24, 194, 129, 162, 128, 1, 77, 147,
+							200, 229, 189, 242, 111, 198, 236, 139, 16, 143, 19, 245, 113, 233, 138, 210}},
+						ProcessedDownwardMessages: 0,
+						HrmpWatermark:             1,
+					},
+					PersistedValidationData: parachaintypes.PersistedValidationData{
+						ParentHead: parachaintypes.HeadData{Data: []byte{1, 0, 0, 0, 0, 0, 0, 0, 1,
+							2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7,
+							8, 9, 0, 1, 2, 48, 246, 146, 178, 86, 226, 64, 9,
+							188, 179, 77, 14, 232, 77, 167, 60, 41, 138, 250, 204, 9, 36, 224, 17, 5, 226, 235,
+							15, 1, 168, 127, 226}},
+						RelayParentNumber: 1,
+						RelayParentStorageRoot: common.MustHexToHash(
+							"0x50c969706800c0e9c3c4565dc2babb25e4a73d1db0dee1bcf7745535a32e7ca1"),
+						MaxPovSize: 2048,
+					},
+				},
+			},
+			isValid: true,
+		},
+	}
+	for name, tt := range tests {
+		tt := tt
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			got, err := validateFromChainState(tt.args.runtimeInstance, tt.args.pov, tt.args.candidateReceipt)
+			if tt.expectedError != nil {
+				require.EqualError(t, err, tt.expectedError.Error())
+			} else {
+				require.NoError(t, err)
+			}
+			require.Equal(t, tt.isValid, got.IsValid())
+			require.Equal(t, tt.want, got)
```

Comment:
this seems fishy since `want` and `got` are both pointers

Reply:
Updated.